File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashMap <Integer, Integer> map = new HashMap<>();

        for(int i = 0; i < nums.length; i++){
            //need to check if the number already exists. 
            if(map.containsKey(nums[i])){
                return true;
            } else {
                map.put(nums[i], 0);
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        hashmap = {}
        duplicate = 0
        for number in nums:
            count = hashmap.get(number, 0)
            if count != 0:
                duplicate = number
                return number
            hashmap[number] = count + 1  
        return  duplicate    